seonshine_frontend/src/components/organims/table/index.tsx

```diff
@@ -0,0 +1,154 @@
+import { ChangeEvent, memo, useMemo, useState } from 'react';
+
+import { ArrowDownward, ArrowUpward } from '@mui/icons-material';
+import {
+  Box,
+  Table as MuiTable,
+  Pagination,
+  Skeleton,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+} from '@mui/material';
+import { Cell, flexRender, getCoreRowModel, Row, SortingState, Updater, useReactTable } from '@tanstack/react-table';
+
+import { CustomColumnDef } from '@/types/table';
+
+const genericMemo: <T>(component: T) => T = memo;
+
+type UserTableProps<T> = {
+  data: T[];
+  columns: CustomColumnDef<T>[];
+  isFetching?: boolean;
+  skeletonHeight?: number;
+  pageCount?: number;
+  page?: (page: number) => void;
+  onClickRow?: (cell: Cell<T, unknown>, row: Row<T>) => void;
+  EmptyText?: string;
+  handleRow?: () => void;
+  onSortingChange?: (sorting: SortingState) => void;
+};
+
+const Table = <T extends object>({
+  data,
+  columns,
+  isFetching = false,
+  skeletonHeight = 28,
+  pageCount,
+  onClickRow,
+  page,
+  EmptyText = 'No Data is found',
+  handleRow,
+  onSortingChange,
+}: UserTableProps<T>) => {
+  const [paginationPage, setPaginationPage] = useState(1);
```

Comment:
the paging must be managed from parent component for save cached. The common component shouldn't save any state. Rename to `currentPage`